iabtcf-decoder/src/main/java/com/iabtcf/utils/BitSetIntIterable.java

```diff
@@ -51,11 +52,14 @@ public boolean isEmpty() {
 
     @Override
     public boolean contains(int value) {
-        try {
-            return bs.get(value);
```

Comment:
I would rather let this method crash in our face if we pass a negative number rather than returning false as if nothing inherently wrong has just happened

Reply:
Why is that an exceptional case? A negative integer is in the domain of ints. IntIterable doesn't claim to only support non-negative integers.

Reply:
there are two cases when it `bs.get(value)`: int is negative or exceeds bs size. I simply introduced `isOutOfBitSetBounds` simply as I don't like when exceptions are used as a flow control as it was here. 

I can create a custom exception of runtime type and simply rethrow it if you prefer

Reply:
> as I don't like when exceptions are used as a flow control.

that's taking it a bit too far. You're checking for bounds twice which is unnecessary.

> I can create a custom exception of runtime type and simply rethrow it if you prefer

The type of exception to throw is not the concern here?

Reply:
@laktech  where do I check twice?

Reply:
please revert this code, there is no value being added here

Reply:
Using exceptions for flow control is an anti pattern so yes, there is value added

Reply:
how was the original code using exceptions as flow control?

Reply:
if int is negative
![image](https://user-images.githubusercontent.com/12062146/78423842-7d0eff80-7669-11ea-8ac4-bb62c8cc59a2.png)

With current impl if you provide negative int then `bs.get()` will throw ex only to be swallowed in catch block and turned into false. What is the point of raising exception if we swallow it? Either we let the ex to be thrown to let the user know what is he doing wrong or simply we can check the int for negatives before calling `bs.get`